Source/Core/Core/AchievementManager.cpp

```diff
@@ -668,17 +734,14 @@ void AchievementManager::DisplayWelcomeMessage()
 
 void AchievementManager::HandleAchievementTriggeredEvent(const rc_client_event_t* client_event)
 {
-  OSD::AddMessage(fmt::format("Unlocked: {} ({})", client_event->achievement->title,
-                              client_event->achievement->points),
-                  OSD::Duration::VERY_LONG,
-                  (rc_client_get_hardcore_enabled(AchievementManager::GetInstance().m_client)) ?
-                      OSD::Color::YELLOW :
-                      OSD::Color::CYAN,
-                  (Config::Get(Config::RA_BADGES_ENABLED)) ?
-                      DecodeBadgeToOSDIcon(AchievementManager::GetInstance()
-                                               .m_unlocked_badges[client_event->achievement->id]
-                                               .badge) :
-                      nullptr);
+  OSD::AddMessage(
+      fmt::format("Unlocked: {} ({})", client_event->achievement->title,
+                  client_event->achievement->points),
+      OSD::Duration::VERY_LONG,
+      (rc_client_get_hardcore_enabled(AchievementManager::GetInstance().m_client)) ?
+          OSD::Color::YELLOW :
+          OSD::Color::CYAN,
+      &AchievementManager::GetInstance().m_unlocked_badges[client_event->achievement->id]);
```

Comment:
For readability I would suggest coalescing multiple calls to `AchievementManager::GetInstance()`. Something like:

```
  auto& instance = AchievementManager::GetInstance();
  OSD::AddMessage(fmt::format("Unlocked: {} ({})", client_event->achievement->title,
                              client_event->achievement->points),
                  OSD::Duration::VERY_LONG,
                  (rc_client_get_hardcore_enabled(instance.m_client)) ? OSD::Color::YELLOW :
                                                                        OSD::Color::CYAN,
                  &instance.m_unlocked_badges[client_event->achievement->id]);
```